addons/account/models/ir_module.py

```diff
@@ -66,7 +67,8 @@ def write(self, vals):
         super().write(vals)
         is_installed = len(self) == 1 and self.state == 'installed'
         if not was_installed and is_installed and not self.env.company.chart_template and self.account_templates:
-            self.env.registry._auto_install_template = next(iter(self.account_templates))
+            templates_by_seq = sorted(self.account_templates.items(), key=lambda kv: kv[1]['sequence'])
+            self.env.registry._auto_install_template = next(iter(templates_by_seq))[0]
```

Comment:
Sort according to the sequence in `_compute_account_templates` so that it is sorted everywhere (i.e. when displaying in the selection field)
It's also clearer to keep everything related in one place.